src/ray/raylet/node_manager.cc

```diff
@@ -576,7 +576,9 @@ void NodeManager::HandleReleaseUnusedBundles(
   std::vector<std::shared_ptr<WorkerInterface>> workers_associated_with_unused_bundles;
   for (const auto &worker_it : leased_workers_) {
     auto &worker = worker_it.second;
-    if (0 == in_use_bundles.count(worker->GetBundleId())) {
+    const auto &bundle_id = worker->GetBundleId();
+    // We need to filter out the workers used by placement group.
+    if (!bundle_id.first.IsNil() && 0 == in_use_bundles.count(bundle_id)) {
```

Comment:
What is `&& 0` for? Doesn't this mean this and statement is always false?

Reply:
!bundle_id.first.IsNil() && (0 == in_use_bundles.count(bundle_id))